knowledge_repo/app/index.py

```diff
@@ -67,23 +67,23 @@ def update_index_required():
         db_session.commit()
 
 
-def update_index():
+def update_index(force=False):
     """
     Initialize the db from a KnowledgeRepository object
     """
 
-    if not update_index_required():
+    if (not force and not update_index_required()) or (force and is_indexing()):
```

Comment:
@danfrankj This logic is now correct. I do think though in retrospect we should probably have just added a `skip_timeouts` option to `update_index_required`; rather than using the force option. Then we wouldn't need to check `is_indexing` multiple times. But if you want to get this in as is, we can clean up the code removing this force option in the future.